src/jamun/model/energy.py

```diff
@@ -339,21 +322,24 @@ def compute_loss(
 
     def noise_and_compute_loss(
         self,
-        x: torch_geometric.data.Batch,
+        x: torch.Tensor,
+        topology: torch_geometric.data.Batch,
         sigma: float | torch.Tensor,
         align_noisy_input: bool,
-    ) -> tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
+    ) -> Tuple[torch.Tensor, torch.Tensor, torch.Tensor]:
```

Comment:
nit: since [pep 585](https://peps.python.org/pep-0585/) type hinting with the standard collections is preferred. They say they will remove `typing.List` etc eventually. ruff has [UP006](https://docs.astral.sh/ruff/rules/non-pep585-annotation/#non-pep585-annotation-up006) for this.